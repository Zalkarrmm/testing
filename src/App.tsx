import './App.css';


function App() {
  return (
    <div className="App">
      React app
      <a href="https://google.com">Link to google</a>
      <button>Click me</button>
      <input type="text" placeholder='input text...'/>
    </div>
  );
}

export default App;
