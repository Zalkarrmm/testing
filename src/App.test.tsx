import { render, screen } from "@testing-library/react"
import App from './App';

describe('React app test', () => {
  test('app page render', () => {
    render(<App/>);
    const linkElement = screen.getByText(/link to google/i)
    const btn = screen.getByRole("button")
    const input = screen.getByPlaceholderText(/input text/i)
    expect(linkElement).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    expect(input).toBeInTheDocument()
    expect(input).toMatchSnapshot()
  })
  test('app', () => {
    render(<App/>);
    const linkElement = screen.getByText(/link to google/i)
    const btn = screen.getByRole("button")
    const input = screen.getByPlaceholderText(/input text/i)
    expect(linkElement).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    expect(input).toBeInTheDocument()
    expect(input).toMatchSnapshot()
  })
})