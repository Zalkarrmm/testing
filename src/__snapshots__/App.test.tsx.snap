// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app test app 1`] = `
<input
  placeholder="input text..."
  type="text"
/>
`;

exports[`React app test app page render 1`] = `
<input
  placeholder="input text..."
  type="text"
/>
`;

exports[`react app render 1`] = `
<input
  placeholder="input text..."
  type="text"
/>
`;
